test/SourceKit/CodeComplete/complete_sort_order.swift

```diff
@@ -205,8 +205,8 @@ func test7() {
   #^CASE_0,caseSensitiveCheck,CaseSensitiveCheck^#
 }
 // CASE_0: Results for filterText: caseSensitiveCheck [
-// CASE_0: caseSensitiveCheck
 // CASE_0: CaseSensitiveCheck
+// CASE_0: caseSensitiveCheck
```

Comment:
This looks like a regression. What happened here?

Reply:
I haven't looked into it, but this looks like a existing bug. The first commit causes this change. Looking at 'CaseSensitiveCheck' case a few lines below, it also suggest non-matching 'caseSensitiveCheck' first.

Maybe exact match mechanism is not working well?

Reply:
The case-sensitive part of this check is actually below with `caseSensitiveCheck.` or in the next case `CaseSensitiveCheck(`.  The reason we currently get the lowercase completion first every time was that it is "local" but the struct is "thismodule" context.  Did that change in this patch? Treating the struct as local here is probably correct, so what happens if you move it outside the function, which is a more realistic case anyway.

Reply:
Ah, yes. I confirmed that they are now both "local".  Previously, because of the `\0`, `}` was ignored, and [the "recovery" logic](https://github.com/apple/swift/pull/28403/files#diff-a5731a934e0c085bf4b84dabd27a6ffdL5583-L5594) closes the `test7` function before the `struct` like:
```swift
func test7() { }
  struct CaseSensitiveCheck {
    var member: Int = 0
  }
  let caseSensitiveCheck = CaseSensitiveCheck()
  <code-completion><eof>
```
So the `CaseSensitiveCheck` and `caseSensitiveCheck` were both global.
Now, they are properly "local" type and "local" variable.

Reply:
SGTM